### PR TITLE
feat: add cover color-changing-cover

### PR DESCRIPTION
**Pull request by enkaito**
This pull request introduces a new (pseudo) semitransparent cover function that provides a method for hiding content in slides. The function changes the text color of elements to a specified color if they contain text. If an element does not contain text, it can use `hide` as a fallback.

This implementation has been successfully tested with the default examples and the specific example provided in issue #112.

### CODE REVIEW

**Comment by OrangeX4** [comment]
Can we achieve translucency by modifying the alpha value instead of fixing a color? We can think about the hacky code that I have as proof of concept below

```typst
#set page(width: 20em, height: auto)

#let update-alpha(constructor: rgb, color, alpha) = constructor(..color.components(alpha: true).slice(0, -1), alpha)

#show par: it => {
  if repr(text.fill).split("(").at(0) == "rgb" {
    set text(fill: update-alpha(text.fill, 50%))
    it
  } else {
    it
  }
}

// ----------------
// content
// ----------------


#set text(rgb("#000000"))

Test

#set text(rgb("#ff0000"))

Test
```

image

**Comment by enkaito** [comment]
How about this?

File: src/utils.typ [hunk 1]
```diff
@@ -802,6 +802,56 @@
...
+  if it.has("body") {
+    return _contains-text(it.body)
+  }
+  if it.has("base") {
+    return _contains-text(it.base)
+  }
+  if it.has("child") {
+    return _contains-text(it.child)
+  }
+  if it.has("children") {
```
Comment (OrangeX4) [inline src/utils.typ:822]: The `table` and `grid` types also have `children`, I'm not sure if this should be included?

Comment (enklht) [reply src/utils.typ:822]: I understand. It might be good to have an option to toggle the behaviour, or exclude those elements (using `util.is-sequence`?)

**Comment by enkaito** [comment]
**Quick Update:** I discovered that the new `alpha-changing-cover` doesn't function perfectly when the text color is modified within the paused items.

File: src/utils.typ (hunk 1 above)
Comment (OrangeX4) [reply src/utils.typ:822]: Since the table has borders, we should use `util.is-sequence` as default, but it can indeed be made into a configuration.

**Comment by enkaito** [comment]
**Update:** The enhanced function now provides improved handling of color-changing texts. This is achieved by utilizing the `context` feature and the `update-alpha` function. This works well with the example I made the other day.

## Note
The decision to use `update-alpha` instead of `transparentize` was deliberate. Unlike `transparentize`, which applies relative changes that can accumulate to cause nested texts to vanish, `update-alpha` allows to specify absolute text opacities. This approach ensures consistent visibility for nested color-changing elements.

**Comment by enkaito** [comment]
While this is slightly tangential to the current PR, I'd like to propose an improvement to the `update-alpha` function:

```typst
#let update-alpha(color, alpha) = color.opacify(100%).transparentize(alpha)
```

This revised implementation can handle all color constructors uniformly. It first ensures full opacity with `opacify(100%)`, then applies the desired transparency using `transparentize(alpha)`.

**Comment by OrangeX4** [comment]
If `update-alpha` is used, we need to consider non-rgb cases, like `#set text(luma(80%))`.

**Comment by enkaito** [comment]
My latest proposal addresses that issue!

**Review by OrangeX4 — APPROVED** [review]

**Comment by OrangeX4** [comment]
Thanks!

**Pull request merged**